### ISSUE TITLE
Manifest handover requires bypassing the private API

### ISSUE THREAD

**Issue by noah-aviel-dove** [DataBiosphere/azul#4399]

**Comment by melainalegaspi** [comment]
@hannes-ucsc :"Spike to consider and document the implications."

**Comment by hannes-ucsc** [comment]
The current (pre #4355) implementation of the `/fetch/manifest/files` endpoint yields a 301 redirect to the non-fetch `/manifest/files` endpoint which ultimately yields a signed URL to the manifest object in the storage bucket. The indirection via the `/manifest/files` endpoint was implemented as a solution for #2875. 

The workaround in #4355 (PR #4378, see referenced commits) causes `/fetch/manifest/files` to return a signed URL immediately, but only on private APIs. This is not a breach of access controls, given that any user with VPN access can use the non-fetch endpoint to get that signed URL. There is, however, an increased risk that a user may inadvertently share the signed URL, a secret. For public APIs, the shared URL points back to Azul, which is not a secret and requires authentication via a bearer token. For private API, this risk is mitigated by two factors: 1) the fact that the set of users with VPN access is very small, and will only include employees of UCSC, Clever Canary and BI; and 2) that the signed URL expires after one hour.

The resolution of this ticket is to replace the FIXME comment with a comment that documents what I described in the above paragraph.
